tests/integration_test/http1_deadline_test.go

```diff
@@ -180,3 +186,96 @@ func TestDeadlinesForLocalBackend(t *testing.T) {
 		})
 	}
 }
+
+// Tests the stream idle timeouts configured via deadline in backend rules for a HTTP/1.x backend.
+func TestIdleTimeoutsForUnaryRPCs(t *testing.T) {
+	t.Parallel()
+
+	testData := []struct {
+		desc           string
+		confArgs       []string
+		reqDuration    time.Duration
+		deadlineToTest ConfiguredDeadline
+		wantErr        string
+	}{
+		// Please be cautious about adding too many time-based tests here.
+		// This can slow down our CI system if we sleep for too long.
+		{
+			desc: "When deadline is NOT specified, default deadline (15s) kicks in and the request fails.",
```

Comment:
Please add comment like following for each case
// used route Timeout = 15s (default), IdleTimeout = 25s
// This 504 is caused by `Timeout`, not by `IdleTimeout`.

Reply:
Done. Looking at them after the weekend, I see why they are confusing. Tried renaming the test desc as well.